src/utils/process.rs

```diff
@@ -421,58 +437,105 @@ where
 
 #[cfg(test)]
 pub mod tests {
-    use super::blame::*;
     use super::*;
 
     use itertools::Itertools;
-
-    pub mod cfg {
-        use std::cell::RefCell;
-
-        #[derive(Debug, PartialEq)]
-        enum TlsState<T> {
-            Some(T),
-            None,
-            Invalid,
+    use std::cell::RefCell;
+    use std::rc::Rc;
+
+    thread_local! {
+        static FAKE_ARGS: RefCell<TlsState<Vec<String>>> = RefCell::new(TlsState::None);
+    }
+
+    #[derive(Debug, PartialEq)]
+    enum TlsState<T> {
+        Once(T),
+        Scope(T),
+        With(usize, Rc<Vec<T>>),
+        None,
+        Invalid,
```

Comment:
Would `Exhausted` or `Used` or `Consumed` be a better name?

(The code here is all quite sophisticated by my Rust standards so I'm really just trying to understand the PR and may not have great suggestions.)